instrumentation/r2dbc-1.0/library/src/main/java/io/opentelemetry/javaagent/instrumentation/r2dbc/v1_0/internal/R2dbcInstrumenterBuilder.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.r2dbc.v1_0.internal;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanKindExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.db.DbClientSpanNameExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.db.SqlClientAttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.net.NetClientAttributesExtractor;
+import io.opentelemetry.instrumentation.api.internal.ConfigPropertiesUtil;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class is internal and is hence not for public use. Its APIs are unstable and can change at
+ * any time.
+ */
+public final class R2dbcInstrumenterBuilder {
+
+  private static final String INSTRUMENTATION_NAME = "io.opentelemetry.r2dbc-1.0";
+
+  private final OpenTelemetry openTelemetry;
+
+  private final List<AttributesExtractor<DbExecution, Void>> additionalExtractors =
+      new ArrayList<>();
+
+  public R2dbcInstrumenterBuilder(OpenTelemetry openTelemetry) {
+    this.openTelemetry = openTelemetry;
+  }
+
+  @CanIgnoreReturnValue
+  public R2dbcInstrumenterBuilder addAttributeExtractor(
+      AttributesExtractor<DbExecution, Void> attributesExtractor) {
+    additionalExtractors.add(attributesExtractor);
+    return this;
+  }
+
+  public Instrumenter<DbExecution, Void> build() {
+
+    return Instrumenter.<DbExecution, Void>builder(
+            openTelemetry,
+            INSTRUMENTATION_NAME,
+            DbClientSpanNameExtractor.create(R2dbcSqlAttributesGetter.INSTANCE))
+        .addAttributesExtractor(
+            SqlClientAttributesExtractor.builder(R2dbcSqlAttributesGetter.INSTANCE)
+                .setStatementSanitizationEnabled(
+                    ConfigPropertiesUtil.getBoolean(
+                        "otel.instrumentation.common.db-statement-sanitizer.enabled", true))
```

Comment:
In general, usually we don't use properties in library instrumentations - instead, we expose a builder method that allows users to configure the setting by themselves. The JDBC library instrumentation is not a good example of that because it utilizes the `Driver` decorator approach - where it's not possible to have a builder to configure a setting.
See the [`CassandraTelemetryBuilder`](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/cassandra/cassandra-4.4/library/src/main/java/io/opentelemetry/instrumentation/cassandra/v4_4/CassandraTelemetryBuilder.java#L37) for an example.